sys-boot/edk2/edk2-2018.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="EFI firmware development environment"
+DESCRIPTION="The EDK II project is the response to the EFI community’s \
+request for a better build and version tracking environment for UEFI and PI \
+development. The main difference between the EDK II to the original EDK is \
+the Enhanced Build Environment of the EDK II. The advantages of the Enhanced \
+Build Environment include:
+
+* Operating System independence
+* Flexibility in choosing the compiler and assembler tools
+* The ability to generate working code using open source build tools and \
+applications
+* Enhanced development and build capability of modules and module packages
+* Use of build configuration tools and data sets to provide flexible process
+* Online source control allows users to contribute code and become participants"
+HOMEPAGE="http://www.tianocore.org/"
+COPYRIGHT="2004-2016 Intel Corporation
+	2008-2010 Apple Inc
+	2011-2015 ARM Limited
+	2014-2015 Linaro Limited
+	2013-2015 Red Hat, Inc."
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/tianocore/edk2/archive/vUDK$portVersion.tar.gz"
+CHECKSUM_SHA256="a37aa6dc7a4326b562ce03e225afd96a0cd1dc847d9aeddfd867ce6b2e7cc7bc"
+SOURCE_DIR="edk2-vUDK$portVersion"
+PATCHES="edk2-$portVersion.patchset"
+DISABLE_SOURCE_PACKAGE="yes"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	edk2 = $portVersion compat >= 0
```

Comment:
```diff
 PATCHES="edk2-$portVersion.patchset"
-DISABLE_SOURCE_PACKAGE="yes"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
+DISABLE_SOURCE_PACKAGE="yes"
+
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+fi
 
 PROVIDES="
-	edk2 = $portVersion compat >= 0
+	edk2$secondaryArchSuffix = $portVersion compat >= 0
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
 	"
```

Remark: for `DISABLE_SOURCE_PACKAGE`, we have about 88 recipes which define this variable, and it seems the majority has it just after `ARCHITECTURES`, but I'm OK with keeping it after `PATCHES` too.